File: rich_game_mechnism/src/com/tw/core/commands/CommandFactory.java
package com.tw.core.commands;

import com.tw.core.Dice;
import com.tw.core.places.Estate;

/**
 * Created by pzzheng on 11/16/16.
 */
public class CommandFactory {
    public static Command BuyTool = new Roll.BuyTool();
    public static Command SelectGift = new Roll.SelectGift();

    public static Command Roll(Dice dice) {
        return new Roll(dice);
    }

    public static Command BuyEstate(Estate estate) {
        return new Roll.BuyEstate(estate);
    }

    public static Command UpgradeEstate(Estate estate) {
        return new Roll.UpgradeEstate(estate);
    }

    public static Command Block(int steps) {
        return new UseBlock(steps);
    }
}

File: ruby/rich_game_ruby/src/commands/command_factory.rb
require_relative 'roll'
require_relative 'buy_estate'
require_relative 'upgrade_estate'
class CommandFactory
  Yes = Command.new
  No = Command.new

  def self.Roll(dice)
    return Roll.new(dice)
  end

  def self.BuyEstate(estate)
    return BuyEstate.new(estate)
  end

  def self.UpgradeEstate(estate)
    return UpgradeEstate.new(estate)
  end

end